File: app/src/main/java/com/websarva/wings/android/dowasurememo/DeleteButton.java
package com.websarva.wings.android.dowasurememo;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteStatement;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.LinearLayout;

import java.util.ArrayList;

//viewを削除するための処理
public class DeleteButton extends LinearLayout implements View.OnClickListener{

    //削除するビューの親ビュー
    LinearLayout activityLinearLayout;

    //削除するビュー
    LinearLayout linearLayout;
    Databasehelper _helper;
    Context context;
    String table;
    int tagId;

    //コンストラクタ
    public DeleteButton
            (Context c,LinearLayout llactivity,LinearLayout parentview,String ta) {
        super(c);
        activityLinearLayout=llactivity;
        linearLayout=parentview;
        context=c;
        table=ta;
    }


    @Override
    public void onClick(View v) {
        activityLinearLayout.removeView(linearLayout);
        tagId=activityLinearLayout.getChildCount()+2;
        _helper=new Databasehelper(context);
        SQLiteDatabase db=_helper.getWritableDatabase();
        String sqlDelete="DELETE FROM "+table+" WHERE _id = ?";
        SQLiteStatement statement=db.compileStatement(sqlDelete);
        statement.bindLong(1,tagId);
        statement.executeUpdateDelete();

    }
}

File: app/src/main/java/com/websarva/wings/android/dowasurememo/DeleteButton.kt
package com.websarva.wings.android.dowasurememo

import android.content.Context
import android.view.View
import android.widget.LinearLayout

//viewを削除するための処理
class DeleteButton(context: Context,
/*削除するビューの親ビュー */
                   var activityLinearLayout: LinearLayout,
/* 削除するビュー*/
                   var linearLayout: LinearLayout?,
                   var table: String) : LinearLayout(context), View.OnClickListener {
    var tagId = 0
    override fun onClick(v: View) {
        activityLinearLayout.removeView(linearLayout)
        tagId = activityLinearLayout.childCount + 2
        val _helper = Databasehelper(context)
        val db = _helper.writableDatabase
        val sqlDelete = "DELETE FROM $table WHERE _id = ?"
        val statement = db.compileStatement(sqlDelete)
        statement.bindLong(1, tagId.toLong())
        statement.executeUpdateDelete()
    }
}